papermill/execute.py

```diff
@@ -136,6 +141,33 @@ def prepare_notebook_metadata(nb, input_path, output_path, report_mode=False):
     return nb
 
 
+def parameterize_path(path, parameters):
+    """Format a path with a provided dictionary of parameters
+
+    Parameters
+    ----------
+    path : string
+       Path with optional parameters, as a python format string
+    parameters : dict
+       Arbitrary keyword arguments to fill in the path
+    """
+    if parameters is None:
+        parameters = {}
+
+    try:
+        return path.format(**parameters)
+    except KeyError:
+        # OrderedDict removes duplicates while maintaining order
+        missing_parameters = OrderedDict.fromkeys(
+            [
+                required_parameter
+                for (_, required_parameter, _, _) in StringFormatter().parse(path)
+                if (required_parameter is not None) and (required_parameter not in parameters)
+            ]
+        ).keys()
+        raise Exception("Missing parameters: {}".format(", ".join(missing_parameters)))
```

Comment:
Maybe `PapermillException` or extend that to `PapermillMissingParameterException`

Reply:
Good idea, will do